File: shared/domain/models/personal.go
package models

import "time"

// PersonalNote は訪問記録に紐づく個人メモ（DeviceDBに格納、他メンバーに共有されない）。
type PersonalNote struct {
	ID            string    `json:"id"`
	VisitRecordID string    `json:"visitRecordId"`
	Note          string    `json:"note"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

// PersonalTag は活動メンバーが個人的に定義するタグ（DeviceDB）。
type PersonalTag struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

// PersonalTagAssignment はタグと訪問記録の紐づけ（DeviceDB）。
type PersonalTagAssignment struct {
	ID            string `json:"id"`
	TagID         string `json:"tagId"`
	VisitRecordID string `json:"visitRecordId"`
}

File: pwa/src/domain/models/personal.ts
// 個人スコープ（DeviceDB 格納、他メンバーに共有されない）のドメインモデル。
// 仕様: docs/wants/08_活動メンバー向けアプリ.md
// 参照実装: shared/domain/models/personal.go

/** 訪問記録に紐づく個人メモ。 */
export interface PersonalNote {
  id: string;
  visitRecordId: string;
  note: string;
  /** ISO 8601 */
  createdAt: string;
  /** ISO 8601 */
  updatedAt: string;
}

/** 活動メンバーが個人的に定義するタグ。 */
export interface PersonalTag {
  id: string;
  name: string;
}

/** タグと訪問記録の紐づけ。 */
export interface PersonalTagAssignment {
  id: string;
  tagId: string;
  visitRecordId: string;
}
